### PR TITLE
Adding EndPoint object. #7232.

### PR DESCRIPTION
**Pull request by cropalato**
https://dev.icinga.org/issues/7232

### CODE REVIEW

File: manifests/object/endpoint.pp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+
+define icinga2::object::endpoint (
+  $ensure                    = 'file',
+  $object_name               = $name,
+  $host                      = undef,
+  $port                      = undef,
+  $log_duration              = undef,
+  $target_dir                = '/etc/icinga2/object/endpoints',
+  $target_file_name          = "${object_name}.conf",
```
Comment (nickchappell) [inline manifests/object/endpoint.pp]: You'll have to set `$target_file_name` to `$name`. The `$name` variable is available and defined when Puppet evaluates the defined type, but it looks like the `$object_name` variable doesn't get 'set' internally until after it tries to create the file resource. When I tested it, it created the file as `/etc/icinga2/conf.d/.conf`.

Comment (smbambling) [inline manifests/object/endpoint.pp]: @nickchappell is correct you can't set variable names from other variables within the same class.

**Comment by cropalato** [comment]
fixed

**Comment by nickchappell** [comment]
Just wanted to give you a heads up that I'll be traveling with family for a little while and may not able to review/merge this until after Christmas.

**Comment by nickchappell** [comment]
I merged this into the develop branch. Thanks @cropalato!

**Pull request closed without merging**